packages/workflow/src/workflow.ts

```diff
@@ -193,6 +260,57 @@ export function scheduleActivity<R>(activityType: string, args: any[], options:
   }) as Promise<R>;
 }
 
+/**
+ * Schedule an activity and run outbound interceptors
+ * @hidden
+ */
+export async function scheduleLocalActivity<R>(
+  activityType: string,
+  args: any[],
+  options: LocalActivityOptions
+): Promise<R> {
+  if (options === undefined) {
+    throw new TypeError('Got empty activity options');
+  }
+
+  let originalScheduleTime = undefined;
+
+  // TODO: is the inital attempt 0 or 1?
+  let attempt = 0;
+
+  for (;;) {
+    const seq = state.nextSeqs.activity++;
+    const execute = composeInterceptors(
+      state.interceptors.outbound,
+      'scheduleLocalActivity',
+      scheduleLocalActivityNextHandler
+    );
+
+    try {
+      return (await execute({
+        activityType,
+        headers: {},
+        options,
+        args,
+        seq,
+        attempt,
+        originalScheduleTime,
+      })) as Promise<R>;
+    } catch (err) {
+      if (err instanceof LocalActivityDoBackoff) {
+        await sleep(tsToMs(err.backoff.backoffDuration));
+        if (typeof err.backoff.attempt !== 'number') {
+          throw new TypeError('Invalid backoff attempt type');
+        }
```

Comment:
Shouldn't this check come before the sleep?

Reply:
Technically this should never happen. It's here because all proto fields are optional but you're right that it'd be more "correct" to sleep after validating the response.